src/components/Tracking.tsx

```diff
@@ -37,39 +38,34 @@ export interface TrackingProps {
 
 const HORIZONTAL = ['bottom', 'top'];
 
-// function infoFromWheel(
-//   event: React.WheelEvent<SVGRectElement>,
-//   axisContext: Record<string, PlotAxisContext>,
-//   plotHeight: number,
-// ) {
-//   const ratio = 1 + event.deltaY * -0.001;
-
-//   // Calculate coordinates
-//   let coordinates: TrackingResult['coordinates'] = {};
-//   const { scale } = axisContext.y;
-//   const yPosition = scale(0);
-//   const y1 =
-//     ratio > 1 ? yPosition * (1 - 1 / ratio) : yPosition * (1 - 1 / ratio);
-
-//   const y2 =
-//     ratio > 1
-//       ? yPosition + (plotHeight - yPosition) / ratio
-//       : plotHeight + (plotHeight - yPosition) * (1 / ratio - 1);
-//   coordinates.y1 = toNumber(scale.invert(y1));
-//   coordinates.y2 = toNumber(scale.invert(y2));
-
-//   return {
-//     event,
-//     coordinates,
-//   };
-// }
-
-function infoFromMouse(
-  event: MouseEvent,
+function infoFromMouse<EventType extends MouseEvent | WheelEvent = MouseEvent>(
+  event: EventType,
   axisContext: Record<string, PlotAxisContext>,
   stateSeries: PlotSeriesState[],
+  plotHeight: number,
+  plotWidth: number,
   target: SVGRectElement,
-): TrackingResult {
+): TrackingResult<EventType> {
+  if (event instanceof WheelEvent) {
+    const ratio = 1 + event.deltaY * -0.001; // Calculate coordinates
+    let newDomain: TrackingResult['newDomain'] = {};
+    for (const key in axisContext) {
+      const { scale, position } = axisContext[key];
+      const total = HORIZONTAL.includes(position) ? plotWidth : plotHeight;
+      const pos = scale(0);
+      const min = ratio > 1 ? pos * (1 - 1 / ratio) : pos * (1 - 1 / ratio);
+
+      const max =
+        ratio > 1
+          ? pos + (total - pos) / ratio
+          : total + (total - pos) * (1 / ratio - 1);
+      newDomain[key] = [
+        toNumber(scale.invert(min)),
+        toNumber(scale.invert(max)),
+      ];
+    }
+    return { event, newDomain };
+  }
```

Comment:
All of this logic should happen inside the `useAxisWheelZoom` hook.
I'm pretty sure that you can implement it using the current axis' min/max values directly instead of the raw pixel width and height.
So, instead of a condition that returns a different object when there a wheel event, it should be enough to add a fourth dictionary in the result with the domains (min and max) of all axes: `domains: { [key]: domain }`

Reply:
done